src/components/Map/Map.tsx

```diff
@@ -349,7 +364,7 @@ export default function Map(props: MapProps): JSX.Element {
         updateFeatureState(highlightStateId, 'highlight', entityOptions?.highlight);
       }
     }
-  }, [entityOptions?.highlight, highlightStateId, updateFeatureState]);
+  }, [entityOptions?.highlight, highlightStateId, updateFeatureState, mapSources]);
```

Comment:
are you seeing any unnecessary dependency warnings? maybe just force a check like `if (!mapSources) return;` same for the geoData dependency below?

Reply:
Didn't see any warnings or lints about this. I do need this to run the effect on mapSources changing. What happened is that this effect was running before the map updates with new sources. If the search resulted in a new source being added and needing to be highlighted, the highlight would not be applied. Same for focus